### ISSUE TITLE
Apps installed inside island but not on mainland are not listed

### ISSUE THREAD

**Issue by haroon-ali** [ukanth/afwall#847]
Hi,
When using Island, the Apps installed inside island but not on mainland are not listed in afwall.
Of course dual apps option is enabled. Apps installed on both or only mainland are listed.

**Comment by ghost** [comment]
Same with shelter

**Comment by johnnyeightpence** [comment]
Is there any solution yet ?

**Comment by gab12** [comment]
I have the same problem. So I installed Afwall in the mainland and in the island. Each Afwall has the list of applications in its container. But I can only apply the rules on one instance. If I apply the rules of afwall in the mainland, the island is blocked, and if I apply the rules of the afall in the island, the appli in mainland are blocked.

**Comment by freebrowser1** [comment]
Issue is still there.

Oneplus 6 stock ROM 10.4 and Magisk root.

I have Afwall+ only in Mainland and even when I enable ALL apps having internet access, including Island itself, apps on the Island do have internet access. 
But when I uncheck wi-fi for only 'Any app' then internet access is lost on the island, despite all apps are individually checked. So thers hould be more 'hidden' apps ?

EDIT: I found a (partial) answer on stackexchange:
https://android.stackexchange.com/questions/228249/afwall-block-entire-internet-in-oasisfeng-island-work-mode

I tried this, enable 'experimental => Multi user', but in 'mainland' I could still not see the apps on the island. This thing is that I want to use the firewall in Mainland only.
